### PR TITLE
Added travis modernizer to update travis configurations

### PR DESCRIPTION
**Pull request by UsamaSadiq**
**Relevant Issue:** [BOM-1507](https://openedx.atlassian.net/browse/BOM-1507)

### Description
Travis Modernizer script performs the following operations on a given `Travis` file:
- Remove Django versions `1.11`, `2.0` && `2.1`
- Remove Python `3.6` from test envs.
- Add Python `3.8` to test envs.

### CODE REVIEW

File: django3_codemods/config_tools/travis_modernizer.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+class TravisModernizer:
+    def __init__(self, file_path):
+        setup_ordered_yaml_parser()
+        self.file_path = file_path
+        self.travis_dict = None
+        self.DJANGO_PATTERN = re.compile("django[0-2][0-1]")  # creates regex to match django111, django20 && django21
+
+    def remove_django_envs(self):
+        with open(self.file_path, 'r') as file:
+            self.travis_dict = yaml.load(file, Loader=yaml.FullLoader)
```
[the remark this answers is not in the document]
Comment (UsamaSadiq) [reply django3_codemods/config_tools/travis_modernizer.py:37]: `SafeLoader` only loads a subset of YAML and misses the indentation of elements so we have to use `FullLoader` and set `default_flow_style=False` to maintain the proper indentation of the Travis elements as in the original file.

File: django3_codemods/config_tools/travis_modernizer.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                env for env in env_list if not self.DJANGO_PATTERN.search(env['env'])
+            ]
+            self.travis_dict['matrix']['include'] = updated_list
+            return self.travis_dict
+
+    def add_python_version(self):
+        with open(self.file_path, 'r') as file:
+            self.travis_dict = yaml.load(file, Loader=yaml.FullLoader)
+            if 'python' in self.travis_dict.keys():
+                self.travis_dict['python'].append('3.8')
```
[the remark this answers is not in the document]
Comment (UsamaSadiq) [reply django3_codemods/config_tools/travis_modernizer.py:49]: I can add a function to remove required python versions from the `Travis` file with a flag to either remove the required version or to replace it with a new version. Let me know if this is what you require for python3.6 test handling.

[the remark this answers is not in the document]
Comment (jmbowman) [reply django3_codemods/config_tools/travis_modernizer.py:49]: Sure, that could work.  Don't spend too much time on it, we can always manually delete the py36 entries if necessary.

File: django3_codemods/config_tools/travis_modernizer.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    def __init__(self, file_path):
+        setup_ordered_yaml_parser()
+        self.file_path = file_path
+        self.travis_dict = None
+        self.DJANGO_PATTERN = re.compile("django[0-2][0-1]")  # creates regex to match django111, django20 && django21
+
+    def remove_django_envs(self):
+        with open(self.file_path, 'r') as file:
+            self.travis_dict = yaml.load(file, Loader=yaml.FullLoader)
+            env_list = self.travis_dict['matrix']['include']
```
[the remark this answers is not in the document]
Comment (UsamaSadiq) [reply django3_codemods/config_tools/travis_modernizer.py:38]: Updated the `remove_django_envs` to handle this condition too.

**Pull request merged**